### PR TITLE
Find the error message when using ethers js signer

### PR DESCRIPTION
**Pull request by pabloruiz55**
See #224

### CODE REVIEW

File: packages/react-app/src/helpers/Transactor.js [hunk 1]
```diff
@@ -118,11 +118,16 @@ export default function Transactor(providerOrSigner, gasPrice, etherscan) {
 
         return result;
       } catch (e) {
-        console.log(e);
-        console.log("Transaction Error:", e.message);
+        let message;
+        if(e.data) // Metamask signer returns data 
+          message = e.data.message;
+        else // Ethers js needs parsing the error
+          message = JSON.parse(JSON.parse(JSON.stringify(e.error)).body).error.message;
```
Comment (azf20) [inline packages/react-app/src/helpers/Transactor.js:125]: This throws on Rinkeby, seems like the error object has a different shape 
image
Maybe if it doesn't have this expected shape we can just print the whole slug? (as I feel like there might be a few variations, based on type of error message...)

**Review by azf20 — CHANGES_REQUESTED** [review]
Works nicely on hardhat! This threw on Rinkeby as the returned error shape is different, maybe try it and set message=slug in the catch?

**Review by azf20 — APPROVED** [review]

File: packages/react-app/src/helpers/Transactor.js (hunk 1 above)
Comment (azf20) [reply packages/react-app/src/helpers/Transactor.js:125]: My bad this is still valid, I had switched back to Master

**Review by azf20 — CHANGES_REQUESTED** [review]
Verified that this is:
Good on hardhat
Ok (without the good descriptor) on Kovan
Throws on Rinkeby

**Comment by pabloruiz55** [comment]
New attempt, tested on localhost, ropsten, kovan, rinkeby and mainnet.
All show pretty much a raw error (as I can't find a proper way to perfectly parse each method on each network, and even then they don't expose the revert reason), except localhost that can show the revert reason.

localhost-mm
ropsten-wallet
localhost-wallet
ropsten-mm
mainnet-mm
mainnet-wallet
rinkeby-mm
rinkeby-wallet
kovan-mm
kovan-wallet

File: packages/react-app/src/helpers/Transactor.js [hunk 2]
```diff
@@ -118,16 +118,10 @@ export default function Transactor(providerOrSigner, gasPrice, etherscan) {
...
-          if(e.data) // Metamask signer returns data 
-            message = e.data.message;
-          else // Ethers js needs parsing the error
-            message = JSON.parse(JSON.parse(JSON.stringify(e.error)).body).error.message;
-        }else{
-          message = e.data ? e.data : JSON.stringify(e);
-        }
+        if (DEBUG) console.log(e);
+        // Accounts for Metamask and default signer on all networks
+        let message = e.data && e.data.message ? e.data.message : e.error && JSON.parse(JSON.stringify(e.error)).body ? JSON.parse(JSON.parse(JSON.stringify(e.error)).body).error.message : e.data ? e.data : JSON.stringify(e);
```
Comment (pabloruiz55) [inline packages/react-app/src/helpers/Transactor.js:123]: I like this better, still not great, but at least it eliminates the chainId-based conditional.

Comment (azf20) [reply packages/react-app/src/helpers/Transactor.js:123]: Nice one!

**Review by azf20 — APPROVED** [review]

**Comment by azf20** [comment]
As you say it's not perfect but great stuff testing all those networks, and as you say this error seems to be down to some combination of the network clients / ethers.js / metamask and it wasn't introduced by this original PR - created https://github.com/austintgriffith/scaffold-eth/issues/241 to track

**Pull request merged**